Detectors/ITSMFT/ITS/workflow/include/ITSWorkflow/CalibratorSpec.h

```diff
@@ -0,0 +1,206 @@
+
+  // Whether or not the fit is good
+  bool success = false;
+};
+
+// Object for storing chip info in TTree
+typedef struct {
+  short int chipID, row, col;
+} PIXEL;
+
+class threshold_map
+{
+ public:
+  threshold_map(const std::map<short int, std::vector<threshold_obj>>& t) : thresholds(t){};
+  std::map<short int, std::vector<threshold_obj>> thresholds;
+};
+
+template <class Mapping>
+class ITSCalibrator : public Task
+{
+ public:
+  // using Mapping=ChipMappingITS;
+  ITSCalibrator();
+  ~ITSCalibrator(); // override = default;
+
+  using ChipPixelData = o2::itsmft::ChipPixelData;
+  o2::itsmft::ChipMappingITS mp;
+
+  void init(InitContext& ic) final;
+  void run(ProcessingContext& pc) final;
+  void endOfStream(EndOfStreamContext& ec) final;
+
+  //////////////////////////////////////////////////////////////////
+ private:
+  TStopwatch mTimer;
+
+  // detector information
+  static constexpr short int NCols = 1024; // column number in Alpide chip
+  static constexpr short int NRows = 512;  // row number in Alpide chip
+  static constexpr short int NLayer = 7;   // layer number in ITS detector
+  static constexpr short int NLayerIB = 3;
+
+  const short int nRUs = o2::itsmft::ChipMappingITS::getNRUs();
+
+  // The number of injections per data value
+  const short int nInj = 50;
+
+  // Number of charges in a threshold scan (from 0 - 50 inclusive)
+  const short int nCharge = 51;
+  // Number of points in a VCASN tuning (from 30 - 70 inclusive)
+  const short int nVCASN = 51;
+  // Number of in a ITHR tuning (from 30 to 100 inclusive)
+  const short int nITHR = 71;
+  // Refernce to one of the above values; updated during runtime
+  const short int* nRange;
+
+  // The x-axis of the correct data fit chosen above
+  short int* x;
+
+  const short int NSubStave[NLayer] = {1, 1, 1, 2, 2, 2, 2};
+  const short int NStaves[NLayer] = {12, 16, 20, 24, 30, 42, 48};
+  const short int nHicPerStave[NLayer] = {1, 1, 1, 8, 8, 14, 14};
+  const short int nChipsPerHic[NLayer] = {9, 9, 9, 14, 14, 14, 14};
+  // const short int ChipBoundary[NLayer + 1] = { 0, 108, 252, 432, 3120, 6480, 14712, 24120 };
+  const short int StaveBoundary[NLayer + 1] = {0, 12, 28, 48, 72, 102, 144, 192};
+  const short int ReduceFraction = 1; // TODO: move to Config file to define this number
+
+  std::array<bool, NLayer> mEnableLayers = {false};
+
+  // Hash tables to store the hit and threshold information per pixel
+  std::map<short int, short int> currentRow;
+  std::map<short int, std::vector<std::vector<short int>>> pixelHits;
+  // std::map< short int, TH2F* > thresholds;
+  //  Unordered vector for saving info to the output
+  std::map<short int, std::vector<threshold_obj>> thresholds;
+  // std::unordered_map<unsigned int, int> mHitPixelID_Hash[7][48][2][14][14]; //layer, stave, substave, hic, chip
+
+  // Tree to save threshold info in full threshold scan case
+  TTree* threshold_tree = new TTree("ITS_calib_tree", "ITS_calib_tree");
```

Comment:
Thanks -- this will be moved elsewhere in the next commit (and initialized with nullptr). We will improve the output so that a ROOT file is opened before each new TTree.